Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs

```diff
@@ -101,5 +111,78 @@ public TestsRepository(TrappistDbContext dbContext, IGlobalUtil util)
+            var testQuestionList = await _dbContext.TestQuestion.Where(x => x.TestId == testId).ToListAsync();
+            var questionList = await _dbContext.Question.Where(x => x.CategoryID == categoryId).Include(y => y.SingleMultipleAnswerQuestion).ThenInclude(x => x.SingleMultipleAnswerQuestionOption).ToListAsync();
+
+            questionList.ForEach(question =>
+            {
+                questionAc = new QuestionAC();
+                questionAc.Question = Mapper.Map<Question, QuestionDetailAC>(question);
+                questionAc.SingleMultipleAnswerQuestion = Mapper.Map<SingleMultipleAnswerQuestion, SingleMultipleAnswerQuestionAC>(question.SingleMultipleAnswerQuestion);
+                if (testQuestionList.Exists(x => x.QuestionId == questionAc.Question.Id && x.TestId == testId))
+                    questionAc.Question.IsSelect = true;
+                questionListAc.Add(questionAc);
+            });
+            return questionListAc;
+        }
+
+        public async Task<string> AddTestQuestionsAsync(List<QuestionAC> questionsToAddTest, int testId)
+        {
+            bool isDeleted = false;
+            TestQuestion testQuestionObj = new TestQuestion();
+            List<TestQuestion> testQuestionList = new List<TestQuestion>();
+            foreach (var questionToAdd in questionsToAddTest)
+            {
+                var question = Mapper.Map<QuestionDetailAC, Question>(questionToAdd.Question);
+                var questionExistInTest = await _dbContext.TestQuestion.FirstOrDefaultAsync(x => x.QuestionId == question.Id);
+                if (questionExistInTest != null && !questionToAdd.Question.IsSelect)
+                {
+                    _dbContext.TestQuestion.Remove(questionExistInTest);
+                    await _dbContext.SaveChangesAsync();
+                    isDeleted = true;
+                }
+
+                else if (questionToAdd.Question.IsSelect)
+                {
+                    testQuestionObj = new TestQuestion();
+                    testQuestionObj.QuestionId = question.Id;
+                    testQuestionObj.TestId = testId;
+
+                    if (!await _dbContext.TestQuestion.AnyAsync(x => x.QuestionId == question.Id && x.TestId == testId))
+                        testQuestionList.Add(testQuestionObj);
+                }
+            }
+            if (!testQuestionList.Any() && !isDeleted)
+                return _stringConstants.NoNewChanges;
+            else
+            {
+                await _dbContext.TestQuestion.AddRangeAsync(testQuestionList);
+                await _dbContext.SaveChangesAsync();
+                return _stringConstants.SuccessfullySaved;
+            }
+        }
+
+        public async Task<TestAC> GetTestDetailsByIdAsync(int testId)
+        {
+            TestAC testAcObject = new TestAC();
+            var test = await _dbContext.Test.FindAsync(testId);
+            testAcObject = Mapper.Map<Test, TestAC>(test);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable 'testAcObject'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1854)